README.md

```diff
@@ -1,2 +1,60 @@
-# comp110-worksheet-5
-Base repository for COMP110 worksheet 5
+# COMP110-Worksheet-5
+
+## Joseph Spaul
+
+
+## (a)
+
+The task that this algorithm performs is for it to compare the length of two lists (list[i] and list[j]) and the 
+compare the values inside the lists. If the values of one list are not equal, but the length of the two lists are
+equal, the algorithm will return True. If the values are equal and the length of the lists are equal, then the 
+algorithm will return False. The same can be said for if the values of i and j are not equal and the length of 
+the lists are not equal too.
+
+## (b)
+
+In the algorithm the second for loop has to be completed in order for the first one to be completed, this is 
+becasue they are nested loops. We take the time it takes for the inner loop to run and multiply it by the time 
+it takes the outer loop to run. 
+
+## (c)
+
+Using i - 1 instead of n - 1 still lets the algorithm be correct because n is the length of the list and i is 
+all of the values in a list. This lets the algorithm run normally. 
+
+## (d)
+
+Because the algorithm will have all of the j values already. So, the time the algorithm takes to work out i and
+j will be halved.
+
+## (e) 
+
+Yes, the algorithm would still be called quadratic. This is becasue the two nested loops are still being used.
+
+## (f) 
+
+O(n log n) [1] 
+
+## (g)
+
+I will calculate the time complexity of the algorithm:
+
+	O(1) + O(n log n) + O(n) + O(1) 
+This can be boiled down to O(n log n) becasue that is the slowest one
+	O(n log n)
+
+## (h)
+
+The algorithm with the complexity O(n log n) will run faster. This is becasue the other algorithm has a complexity
+of O(n^2) and O(n log n) is faster than O(n^2).
+
+## (i)
+
+The faster the algorithm, the harder it is to make. So, if the programmer doesnt have much time, or the ability, 
+they can use a slower algorithm.
```

Comment:
It's not necessarily the case that a faster algorithm is harder to make -- but in general this is a fair point